src/main/webapp/database-tables.js

```diff
@@ -80,12 +80,55 @@ function createTable(name) {
 }
 
 function mainLoad(){
-  showDatabase();
+  console.log("main");
   login();
+  showReason();
 }
 
 function login() {
   fetch("/login").then(response => response.json()).then((user) => {
     document.getElementById("user").innerText = user;
+    var currentUser = user;
+    if (currentUser[0] == "Stranger"){
+        window.location.assign("https://accounts.google.com/signin/v2/identifier?");
+        return;
+    }
+
+    var usersWithAccess = ["jiaxinz@google.com","gagomez@google.com",/*"test@example.com",*/"hilakey@google.com","sasymer@google.com","williamdc@google.com"];
+    console.log(currentUser[0]);
+    console.log("current user");
+    for (var i =0; i < usersWithAccess.length; i++){
+        if (currentUser[0] == usersWithAccess[i]){
+            return;
+        }
+    }
+    //TODO: Make sure this url is for the deployed version
+    window.location.assign("https://8080-25c0ac2a-87ce-4126-a0a0-8231fedddb09.us-central1.cloudshell.dev/denied.html");
```

Comment:
You could use window.location.hostname to find the right url. E.g. window.location.hostname + "/denied.html"

Reply:
This is very helpful, done!